### PR TITLE
Reorganize the GraphQL Spec

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Purpose

Fixes: [#4466](https://github.com/ballerina-platform/ballerina-standard-library/issues/4466)

## Examples
N/A

## Checklist
- [x] Linked to an issue
- [ ] ~Updated the changelog~
- [ ] ~Added tests~
- [x] Updated the spec
- [ ] ~Checked native-image compatibility~
- [x] No `commons` package changes (if there are any, please update the GraphQL version in [GraphQL tools](https://github.com/ballerina-platform/graphql-tools))

### CODE REVIEW

File: docs/spec/spec.md [hunk 1]
```diff
@@ -1538,1435 +1454,1559 @@ function handleErrors(graphql:ClientError clientError) {
...
-### 10.1 Service Configuration
-
-The configurations stated in the `graphql:ServiceConfig`, are used to change the behavior of a particular GraphQL service. These configurations are applied to the service.
+The configurations stated in the `graphql:InterceptorConfig`, are used to change the behavior of a particular GraphQL interceptor.
 
-This annotation consists of the following fields.
+#### 7.3.1 Scope Configuration
 
-#### 10.1.1 Max Query Depth
+The field `global` is used to configure the scope of the interceptor. If the `global` field is set as `true`, the interceptor will be applied to each field and subfield of the service. If the flag is set as `false`, the interceptor will be applied only to the fields of the type, but not to the subfields of the type. By default, the `global` flag is set as `true`.
```
Comment (MohamedSabthar) [inline docs/spec/spec.md:1716]:
```suggestion
The field `global` is used to configure the scope of the interceptor. If the `global` field is set as `true`, the interceptor will be applied to each field and subfield of the service. If the flag is set as `false`, the interceptor will be applied only to the top level fields of the Query, Mutation and Subscription types, but not to the subfields of them. By default, the `global` flag is set as `true`.
```

File: docs/spec/spec.md [hunk 2]
```diff
@@ -653,20 +656,33 @@ The `String` type is represented using the `string` type in Ballerina. It can re
 The `Boolean` type is represented using the `boolean` type in Ballerina.
 
 #### 4.1.5 ID
-The `ID` type is represented using the `@graphql:ID` annotation in Ballerina.
 
-###### Example: Scalar type ID 
+The `ID` scalar type represents a unique identifier, often used to refetch an object or as the key for a cache. The `ID` type is serialized in the same way as a `String`; however, it is not intended to be human-readable.
```
Comment (aashikam) [inline docs/spec/spec.md:660]:
```suggestion
The `ID` scalar type represents a unique identifier, often used to re-fetch an object or as the key for a cache. The `ID` type is serialized in the same way as a `String`; however, it is not intended to be human-readable.
```

File: docs/spec/spec.md [hunk 3]
```diff
@@ -1159,133 +1175,33 @@ public enum Status {
...
-###### Example: Single File Upload Request
-
-```shell
-curl localhost:9090/graphql \
-    -F operations='{ "query": "mutation($file: Upload!) { fileUpload(file: $file) { link } }", "variables": { "file": null } }' \
-    -F map='{ "0": ["variables.file"] }' \
-    -F 0=@file1.png
-```
-
-###### Example: Multiple File Upload Request
```
Comment (aashikam) [inline docs/spec/spec.md:1178]:
```suggestion
```

File: docs/spec/spec.md [hunk 4]
```diff
@@ -1159,133 +1175,33 @@ public enum Status {
...
 
-```shell
-curl localhost:9090/graphql \
-    -F operations='{ "query": "mutation($file: [Upload!]) { filesUpload(file: $file) { link } }", "variables": { "file": [null, null] } }' \
-    -F map='{ "0": ["variables.file.0"], "1": ["variables.file.1"]}' \
-    -F 0=@file1.png
-    -F 1=@file2.png
-```
```
Comment (aashikam) [inline docs/spec/spec.md:1179]:
```suggestion
```

File: docs/spec/spec.md [hunk 5]
```diff
@@ -1538,1435 +1454,1559 @@ function handleErrors(graphql:ClientError clientError) {
...
 ```ballerina
-@graphql:ServiceConfig {
-    interceptors: [new Interceptor1(), new Interceptor2()]
-}
-service on new graphql:Listener(9090) {
-    // ...
-}
-```
+graphql:FileUserStoreConfig config = {};
+final http:ListenerFileUserStoreBasicAuthHandler handler = new (config);
```
Comment (MohamedSabthar) [inline docs/spec/spec.md:1916]:
```suggestion
final http:ListenerFileUserStoreBasicAuthHandler handler = new;
```

File: docs/spec/spec.md (hunk 1 above)
Comment (ThisaruGuruge) [reply docs/spec/spec.md]: Fixed with 1ccb2c5cc1f0c73c8af646636a5ac78b59f09ce3

File: docs/spec/spec.md (hunk 2 above)
Comment (ThisaruGuruge) [reply docs/spec/spec.md]: Fixed with 1ccb2c5cc1f0c73c8af646636a5ac78b59f09ce3

File: docs/spec/spec.md [hunk 6]
```diff
@@ -1542,1435 +1496,1568 @@ function handleErrors(graphql:ClientError clientError) {
...
+```ballerina
+public function main() returns error? {
+    graphql:Client graphqlClient = check new ("localhost:9090/graphql",
+        auth = {
+            username: "ballerina",
+            issuer: "wso2",
+            audience: ["ballerina", "ballerina.org", "ballerina.io"],
+            keyId: "5a0b754-895f-4279-8843-b745e11a57e9",
+            jwtId: "JlbmMiOiJBMTI4Q0JDLUhTMjU2In",
+            customClaims: {"scp": "admin"},
```
Comment (MohamedSabthar) [inline docs/spec/spec.md:2246]:
```suggestion
            customClaims: {scp : "admin"},
```

File: docs/spec/spec.md [hunk 7]
```diff
@@ -1542,1435 +1496,1568 @@ function handleErrors(graphql:ClientError clientError) {
...
-3. Resolver: name
-5. Connection closed!
+```ballerina
+graphql:Client graphqlClient = check new ("localhost:9090/graphql",
+    auth = {
+        refreshUrl: "https://localhost:9445/oauth2/token",
+        refreshToken: "24f19603-8565-4b5f-a036-88a945e1f272",
+        clientId: "FlfJYKBD2c925h4lkycqNZlC2l4a",
+        clientSecret: "PJz0UhTJMrHOo68QQNpvnqAY_3Aa",
+        scopes: ["admin"],
```
Comment (MohamedSabthar) [inline docs/spec/spec.md:2320]:
```suggestion
        scopes: "admin",
```

File: docs/spec/spec.md [hunk 8]
```diff
@@ -1538,1435 +1454,1559 @@ function handleErrors(graphql:ClientError clientError) {
...
+```ballerina
+public function main() returns error? {
+    graphql:Client graphqlClient = check new ("localhost:9090/graphql",
+        auth = {
+            username: "ballerina",
+            issuer: "wso2",
+            audience: ["ballerina", "ballerina.org", "ballerina.io"],
+            keyId: "5a0b754-895f-4279-8843-b745e11a57e9",
+            jwtId: "JlbmMiOiJBMTI4Q0JDLUhTMjU2In",
+            customClaims: {"scp": "admin"},
```
Comment (MohamedSabthar) [inline docs/spec/spec.md:2246]:
```suggestion
            customClaims: {scp : "admin"},
```

**Review by MohamedSabthar — APPROVED** [review]

**Pull request merged**